### ISSUE TITLE
[BUG] STIR/SHAKEN Date management issue

### ISSUE THREAD

**Issue by abieuzent** [OpenSIPS/opensips#3427]
**OpenSIPS version you are running**
Version: opensips 3.4.6 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, Q_MALLOC, F_MALLOC, HP_MALLOC, DBG_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: 40206eeef
main.c compiled on  with gcc 10

**Describe the bug**
sometime stir_shaken_verify function return an error -6

**Exemple**
INVITE sip:+33377005122@manifone.com;user=phone SIP/2.0
Via: SIP/2.0/UDP 80.10.140.50:5060;branch=z9hG4bK03ccu63098v56dmd0tp0.1;initialinvite=yes
From: "De 0380712340"<sip:+33380712340@orange-multimedia.fr;user=phone>;tag=SDkbm5b02-1327565577-1720101378471-
To: <tel:+33377005122>
Call-ID: SDkbm5b02-fdd2a8b42f93d2f8de5d97130c0db1b2-v300g000I0
CSeq: 1057230036 INVITE
Contact: <sip:80.10.140.50:5060;transport=udp>
P-Asserted-Identity: "De 0380712340"<sip:+33380712340@orange-multimedia.fr;user=phone>
Privacy: none
P-Access-Network-Info: GSTN;operator-specific-GI="552153500";network-provided
P-Early-Media: supported
Allow: INVITE,ACK,CANCEL,BYE,OPTIONS,UPDATE,PRACK
Accept: application/sdp
Max-Forwards: 67
Content-Type: application/sdp
Content-Length: 252
Date: Thu, 04 Jul 2024 15:56:18 GMT
Identity: eyJhbGciOiJFUzI1NiIsInBwdCI6InNoYWtlbiIsInR5cCI6InBhc3Nwb3J0IiwieDV1IjoiaHR0cHM6Ly9hcGkubWFuLWJwY28uZnIvY2VydHMvRlJURTAwLzc1MDk2MjQxNjNiMjIxYjAuY2VyIn0.eyJhdHRlc3QiOiJBIiwiZGVzdCI6eyJ0biI6WyIzMzM3NzAwNTEyMiJdfSwiaWF0IjoxNzIwMTAxMzc4LCJvcmlnIjp7InRuIjoiMzMzODA3MTIzNDAifSwib3JpZ2lkIjoiMTc3MzQwMGItZWZhZi01NWRmLThlZWEtYzM4YjU3MmE4NmRlIn0.zYBAdNETPO4T8qPZq-XsrkDBshOW2F4knH-CjhqeKsOjdIW7Cux9OOAyepOqwX8_01ONYXids4ZpR34-CA2egQ;info=<https://api.man-bpco.fr/certs/FRTE00/7509624163b221b0.cer>;alg=ES256;ppt=shaken

call received on opensips at 15:56:18 (so Date header match the opensips current date)

When we check at the value in signature, it's correct too :
{"attest":"A","dest":{"tn":["33377005122"]},"iat":1720101378,"orig":{"tn":"33380712340"},"origid":"1773400b-efaf-55df-8eea-c38b572a86de"}
The value of the 'iat' parameter corresponds to the date: Thu Jul 04 2024 15:56:18 GMT+0200 (heure d’été d’Europe centrale)

**Expected behavior**
there is no reason to return error -6

**Stirshaken configuration**
modparam("stir_shaken", "require_date_hdr", 0)
modparam("stir_shaken", "verify_date_freshness", 7200)

**OS/environment information**
 - Debian 11
 - OpenSIPS installation:  debs

**Comment by bogdan-iancu** [comment]
Indeed, the `iat` translates to `Thursday, July 4, 2024 1:56:18 PM` GMT. What is the log matching this `-6` retcode?
In code there is an 
```
			LM_NOTICE("Date header timestamp diff exceeds local policy "
			    "(diff: %llds, verify-freshness: %ds)\n",
```
for the case.

**Comment by abieuzent** [comment]
one new exemple :
Jul  5 19:10:12 lbsip-owf-did01 opensips[1124]: NOTICE:stir_shaken:w_stir_verify: Date header timestamp diff exceeds local policy (diff: -7200s, verify-freshness: 60s)

Identity: eyJhbGciOiJFUzI1NiIsInBwdCI6InNoYWtlbiIsInR5cCI6InBhc3Nwb3J0IiwieDV1IjoiaHR0cHM6Ly9hcGkubWFuLWJwY28uZnIvY2VydHMvRlJURTAwLzc1MDk2MjQxNjNiMjIxYjAuY2VyIn0.eyJhdHRlc3QiOiJBIiwiZGVzdCI6eyJ0biI6WyIzMzk0ODAwOTc1OSJdfSwiaWF0IjoxNzIwMTk5NDEyLCJvcmlnIjp7InRuIjoiMzM2NjMzMTcyMjkifSwib3JpZ2lkIjoiMWY0NmRjYzktODg3NS01N2YzLWE0MTUtNDQ1MDk3OTA0MTE1In0.W2uLFgGmP_NhVSHPGkh5-FNbYv8oc1T8iuibpEBu8uitP4XhDJbvzZ7MhmX16lSnS1LkcA-xLoWliUfHdAFL4A;info=<https://api.man-bpco.fr/certs/FRTE00/7509624163b221b0.cer>;alg=ES256;ppt=shaken

current stir_shaken parameters :
modparam("stir_shaken", "require_date_hdr", 0)
modparam("stir_shaken", "verify_date_freshness", 7200)

**Comment by liviuchircu** [comment]
Hi @abieuzent!  Is it possible you set the `verify_date_freshness` by mistake back to `60`, maybe _below_ your loadmodule, or possibly included by mistake via some hard-to-spot `include_file "routes.cfg"` statement?

```
...
# hidden, way below
modparam("stir_shaken", "verify_date_freshness", 60)
```

Alternatively, maybe the service hasn't been restarted with the new `opensips.cfg` (value: `7200`) and it's still using an old `60` value?  I'm saying all of this because no other code sets the `verify_date_freshness` except for the modparam, and the log says the current value was: `60`, **not** `7200` as we expect!  So we must clear these assumptions first before going deeper.

**Comment by abieuzent** [comment]
you are probably right, I remember going from 60 to 7200, but only launching a "reload". I restarted the proxy this morning, since then I haven't had any errors.

Sorry about the noise...
